lib/logstash/codecs/json_lines.rb

```diff
@@ -51,31 +70,21 @@ def encode(event)
   def flush(&block)
     remainder = @buffer.flush
     if !remainder.empty?
-      parse(@converter.convert(remainder), &block)
+      parse_json(@converter.convert(remainder), &block)
     end
   end
 
   private
 
-  # from_json_parse uses the Event#from_json method to deserialize and directly produce events
-  def from_json_parse(json, &block)
-    LogStash::Event.from_json(json).each { |event| yield event }
-  rescue LogStash::Json::ParserError => e
-    @logger.warn("JSON parse error, original data now in message field", :error => e, :data => json)
-    yield LogStash::Event.new("message" => json, "tags" => ["_jsonparsefailure"])
+  def parse_json(json)
+    events_from_json(json, targeted_event_factory).each { |event| yield event }
+  rescue => e
+    @logger.warn("JSON parse error, original data now in message field", message: e.message, exception: e.class, data: json)
```

Comment:
a `@logger.error` message is logged from the json codec, maybe we should align logging levels ...